src/software/thunderscope/play/playinfo_widget.py

```diff
@@ -35,42 +34,42 @@ def __init__(self, buffer_size: int = 1) -> None:
 
     def refresh(self) -> None:
         """Update the play info widget with new play information"""
-        playinfo = self.playinfo_buffer.get(block=False, return_cached=False)
+        playinfo = self.playinfo_buffer.get(
+            block=False, return_cached=False
+        ).GetOptions()
 
         # Updating QTableWidget could be expensive, so we only update if there is new data
         if playinfo is None or playinfo == self.last_playinfo:
             return
 
         self.last_playinfo = playinfo
 
-        play_info_dict = MessageToDict(playinfo)
-
         robot_ids = []
         tactic_fsm_states = []
         tactic_names = []
         play_name = []
 
-        if "robotTacticAssignment" not in play_info_dict:
+        if "robotTacticAssignment" not in playinfo:
             return
 
         num_rows = max(
-            len(play_info_dict["robotTacticAssignment"]),
-            len(play_info_dict["play"]["playState"]),
+            len(playinfo.robotTacticAssignment()),
+            len(playinfo.play.playState()),
```

Comment:
I think this should be `len(playinfo.play.play_state)`, you should check these two comments.